dali/operators/geometry/affine_transforms/scale_transform.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/geometry/affine_transforms/transform_base_op.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(ScaleTransform)
+  .DocStr(R"code(Produces a scale affine transform matrix.
+
+If another transform matrix is passed as an input, the operator applies scaling to the matrix provided.
+
+.. note::
+    The output of this operator can be fed directly to the ``MT`` argument of ``CoordTransform`` operator.
+)code")
+  .AddArg(
+    "scale",
+    R"code(The scale factor, per dimension.
+
+The number of dimensions of the transform is inferred from this argument.)code",
+    DALI_FLOAT_VEC, true)
+  .AddOptionalArg<std::vector<float>>(
+    "center",
+    R"code(The center of the scale operation.
+
+If provided, the number of elements should match the one of ``scale`` argument.)code",
+    nullptr, true)
+  .NumInput(0, 1)
+  .NumOutput(1)
+  .AddParent("TransformAttr");
+
+/**
+ * @brief Scale transformation.
+ */
+class ScaleTransformCPU
+    : public TransformBaseOp<CPUBackend, ScaleTransformCPU> {
+ public:
+  using SupportedDims = dims<1, 2, 3, 4, 5, 6>;
+
+  explicit ScaleTransformCPU(const OpSpec &spec) :
+      TransformBaseOp<CPUBackend, ScaleTransformCPU>(spec),
+      scale_("scale", spec),
+      center_("center", spec) {
+    assert(scale_.IsDefined());
+  }
+
+  template <typename T, int mat_dim>
+  void DefineTransforms(span<affine_mat_t<T, mat_dim>> matrices) {
+    constexpr int ndim = mat_dim - 1;
+    assert(matrices.size() == static_cast<int>(scale_.size()));
+    for (int i = 0; i < matrices.size(); i++) {
+      auto &mat = matrices[i];
+      auto scale = scale_[i];
+      mat = affine_mat_t<T, mat_dim>::identity();
+      for (int d = 0; d < ndim; d++) {
+        mat(d, d) = scale[d];
+      }
+
+      if (center_.IsDefined()) {
+        auto center = center_[i];
+        for (int d = 0; d < ndim; d++) {
+          mat(d, ndim) = center[d] * (T(1) - scale[d]);
+        }
+      }
+    }
+  }
+
+  void ProcessArgs(const OpSpec &spec, const workspace_t<CPUBackend> &ws) {
+    int repeat = IsConstantTransform() ? 0 : nsamples_; 
+    scale_.Read(spec, ws, repeat);
+    ndim_ = scale_[0].size();
+    center_.Read(spec, ws, repeat);
```

Comment:
I think you should read only when it is defined, not the opposite. Reverse L84 and L85